job-status-service/src/main/java/au/org/aodn/aws/wps/lambda/JobStatusServiceRequestHandler.java

```diff
@@ -333,4 +388,131 @@ private String getStatusDescription(StatusType currentStatus) {
         }
         return UNKNOWN_STATUS_DESCRIPTION;
     }
+
+
+    private String getRequestSummary(S3Object requestFileObject) {
+
+        //  Currently just returns the content of the request file.
+        //  Can be expanded to provide a richer HTML summary of the request for
+        //  admin purposes (possible including links to logs etc).
+        try {
+
+            if (requestFileObject != null) {
+
+                S3ObjectInputStream requestInputStream = requestFileObject.getObjectContent();
+                StringBuilder requestStringBuilder = new StringBuilder();
+                int chunkSizeBytes = 1024;
+                byte[] inBytes = new byte[chunkSizeBytes];
+                int bytesRead;
+                while((bytesRead = requestInputStream.read(inBytes)) > 0) {
+                    String inChunk = new String(inBytes, 0, bytesRead -1);
+                    requestStringBuilder.append(inChunk);
+                }
+
+                return requestStringBuilder.toString();
+            } else {
+                LOGGER.error("Request S3 object null.");
+            }
+
+        } catch (IOException ioex) {
+            LOGGER.error("Unable to read request file. Bucket [" + configS3Bucket + "]: " + ioex.getMessage(), ioex);
+        }
+
+        return null;
+    }
+
+
+    private String generateQueueViewHTML(AWSBatch batchClient, String queueName) throws IOException, TemplateException {
+        List<JobSummary> waitingJobSummaries = AWSBatchUtil.getWaitingJobs(batchClient, queueName);
+        List<JobSummary> runningJobSummaries = AWSBatchUtil.getRunningJobs(batchClient, queueName);
+        List<JobSummary> completedJobSummaries = AWSBatchUtil.getCompletedJobs(batchClient, queueName);
+
+
+        List<JobDetail> waitingJobDetails = null;
+        LOGGER.info("Waiting jobs: " + waitingJobSummaries.size());
+        if(waitingJobSummaries.size() > 0) {
+
+            ArrayList<String> jobIds = new ArrayList<>();
+            for (JobSummary summary : waitingJobSummaries) {
+                jobIds.add(summary.getJobId());
+            }
+
+            waitingJobDetails = AWSBatchUtil.getJobDetails(batchClient, jobIds);
+        }
+
+        List<JobDetail> runningJobDetails = null;
```

Comment:
Basically a repeat of above with different objects - would it be better to reuse a single function to reduce the risk of this code dievreging/needing to make the same change in several places.